test/cli.js

```diff
@@ -42,4 +44,75 @@ describe('cli', function () {
       access(dest + 'bartest_0.0.1_amd64.deb', done)
     })
   })
+
+  describe('with duplicate dependencies', function (test) {
+    var dest = 'test/fixtures/out/tjaq/'
+    var config = 'test/fixtures/config.json'
+
+    // Default options (from src/installer.js)
+    var defaults = {
+      depends: ['gvfs-bin',
+        'libgconf2-4',
+        'libgtk2.0-0',
+        'libnotify4',
+        'libnss3',
+        'libxtst6',
+        'xdg-utils'],
+      recommends: ['pulseaudio | libasound2'],
+      suggests: ['gir1.2-gnomekeyring-1.0',
+        'libgnome-keyring0',
+        'lsb-release'],
+      enhances: [],
+      preDepends: []
+    }
+
+    before(function (done) {
+      spawn('./src/cli.js', [
+        '--src', 'test/fixtures/app-with-asar/',
+        '--dest', dest,
+        '--arch', 'i386',
+        '--config', config
+      ], done)
+    })
+
+    after(function (done) {
+      fs.remove(dest, done)
+    })
+
+    it('removes duplicate dependencies', function (done) {
+      access(dest + 'footest_0.0.1_i386.deb', function () {
+        var dpkgDebCmd = 'dpkg-deb -f footest_0.0.1_i386.deb ' +
+          'Depends Recommends Suggests Enhances Pre-Depends'
+        child.exec(dpkgDebCmd, { cwd: dest }, function (err, stdout, stderr) {
+          if (err) return done(err)
+          if (stderr) return done(new Error(stderr.toString()))
+
+          // object with both user and default dependencies based on src/installer.js
+          fs.readJson(config, function (err, configObj) {
+            if (err) return done(err)
+
+            var baseDependencies = {}
+            baseDependencies['Depends'] = _.sortBy(_.union(defaults.depends, configObj.depends))
+            baseDependencies['Recommends'] = _.sortBy(_.union(defaults.recommends, configObj.recommends))
+            baseDependencies['Suggests'] = _.sortBy(_.union(defaults.suggests, configObj.suggests))
+            baseDependencies['Enhances'] = _.sortBy(_.union(defaults.enhances, configObj.enhances))
+            baseDependencies['Pre-Depends'] = _.sortBy(_.union(defaults.preDepends, configObj.preDepends))
+
+            // Creates object based on stdout (values are still strings)
+            var destDependencies = _.fromPairs(_.chunk(_.initial(stdout.split(/\n|:\s/)), 2))
+            // String values are mapped into sorted arrays
+            destDependencies = _.mapValues(destDependencies, function (value) {
+              if (value) return _.sortBy(value.split(', '))
+            })
+
+            if (_.isEqual(baseDependencies, destDependencies)) {
+              done()
+            } else {
+              done(new Error('There are duplicate dependencies'))
+            }
+          })
+        })
+      })
+    })
+  })
```

Comment:
Ugh, more tests with a lot of boilerplate.

Reply:
I know, at least this test is for the `cli` so it's not as crowded. Also, we never had a test for loading configurations from a `json` file.
I'm not really sure how to fix our tests, besides getting rid of some duplicate tests we have. Nonetheless, that should be a different PR.

Reply:
Filed https://github.com/unindented/electron-installer-debian/issues/115.